rs/execution_environment/src/execution_environment.rs

```diff
@@ -1622,9 +1623,30 @@ impl ExecutionEnvironment {
                 )
             }
             CanisterMessageOrTask::Message(CanisterMessage::Request(request)) => {
+                if let Some(metadata) = &request.metadata {
+                    let message_type_label = "request";
+                    let call_tree_depth_label = &metadata.call_tree_depth().to_string();
+                    let latency = current_time()
+                        .saturating_duration_since(*metadata.call_tree_start_time())
```

Comment:
IIRC the call tree start time is the time when the message at the root of the call tree started execution. Regardless, it has to do with the call tree, not with the individual message being executed. So a message enqueued in the same block (i.e. at the same time) when it's executed could have a call tree start time that is arbitrarily old.